enhancements/oc/upgrade_kubernetes.md

```diff
@@ -25,32 +25,48 @@ superseded-by:
-   is a handful query (don’t forget to replace versions from previous query accordingly). For those you just need
-   to verify what kind of patches were applied to previous version, create a spreadsheet
-   (see https://docs.google.com/spreadsheets/d/1LIqRpdSnyTkhD5Dhw8H5QDwd9U98APoZv4lUa5XAfa8/edit for example)
-   and decide whether we still need or not a patch.
-5. In kubernetes repository:
-   1. Add replace dependency to openshift/api and openshift/client-go pointing at latest SHA from that repo, eg.
+4. Open a PR in openshift/release to add this new branch of openshift/kubernetes, [similar to this](https://github.com/openshift/release/pull/6349)
+   Usually, you can copy the last version's `release/ci-operator/config/openshift/kubernetes/*` file to a new file that reflects the new branch name, 
+   verify the go version used, then `make jobs` to generate the new job. 
+5. Pick carry patches. From each openshift/kubernetes-repository, `git log --no-merges --oneline openshift/oc-4.4-kubernetes-1.17.0`
+   is a handful query where `openshift` is the name of the remote pointing to `openshift/kubernetes-repo` 
+   (replace versions from previous query accordingly). For the `UPSTREAM` commits, you need
+   to verify what kind of patches were applied to the last kubernetes bump (1.16.2 here) and create a 
+   [spreadsheet similar to this](https://docs.google.com/spreadsheets/d/1VQw_B2Nfqg9ILKvoNLK0YQdiM2LUidOX7Q-wRKBlrSE/edit?usp=sharing)
+   to decide whether we still need a patch or not.  
+6. For each repository (apimachinery, client-go, cli-runtime, kubectl), open A PR with the picked commits from the
+   spreadsheet against the oc-A.B-kubernetes-X.Y.Z branch.
+7. In openshift/kubernetes repository, check out the new oc-A.B-kubernetes-X.Y.X branch and:
+   1. Add the replace dependency for openshift/api and openshift/client-go pointing at latest SHA from that repo, eg.
       ```
       github.com/openshift/api => github.com/openshift/api master
       ```
-   2. Run `go list -m all` which will turn above into something like:
+   2. Run `go list -m all` which will turn above line in go.mod into something like:
       ```
       github.com/openshift/api v3.9.1-0.20190822120857-58aab2885e38+incompatible // indirect
       ```
-   3. Copy changes from apimachinery, client-go, cli-runtime and kubectl into staging directory,
-      and use git add *.go because we care only about go files.
+   3. Copy and paste the changes from apimachinery, client-go, cli-runtime and kubectl PRs above into kubernetes/staging/src/k8s.io/ directory,
+      and use git add *.go because we care only about go files.  This is a manual step, You can use curl like so for each file to copy file 
+      changes from your PRs above:
+      ```
+      cd staging/src/k8s.io/repo
+      curl -o path/to/file https://raw.githubusercontent.com/.....
+      ```
```

Comment:
I actually did this instead, but maybe the curl example is better:
1. cp go.mod /tmp/go.mod -> will restore 
1. update go.mod for each fork something like 
    `k8s.io/client-go => github.com/openshift/kubernetes-client-go oc-4.4-kubernetes-1.17.0`
2. go mod vendor
3.  cp -R vendor/k8s.io/client-go/* staging/src/k8s.io/client-go/. && rm -rf vendor/k8s.io/client-go
4. cp /tmp/go.mod go.mod (restore go.mod back to original)

Reply:
That seems reasonable, maybe we could add a carry script that would do this for us.